pkg/api/vlabs/validate.go

```diff
@@ -295,8 +295,13 @@ func (o *OpenShiftConfig) Validate() error {
 
 // Validate implements APIObject
 func (m *MasterProfile) Validate(o *OrchestratorProfile) error {
-	if o.OrchestratorType == OpenShift && m.Count != 1 {
-		return errors.New("openshift can only deployed with one master")
+	if o.OrchestratorType == OpenShift {
+		if m.Count != 1 {
+			return errors.New("openshift can only deployed with one master")
+		}
+		if (m.VnetSubnetID != "" && m.FirstConsecutiveStaticIP == "") || (m.FirstConsecutiveStaticIP != "" && m.VnetSubnetID == "") {
```

Comment:
1) I think it should be allowed to specify/override FirstConsecutiveStaticIP in the non-existing vnet case.
2) We default FirstConsecutiveStaticIP somewhere - bearing that in mind, does this work correctly?  Perhaps we should not default FirstConsecutiveStaticIP when VnetSubnetID is set?
3) Needs updated unit test.

Reply:
> We default FirstConsecutiveStaticIP somewhere - bearing that in mind, does this work correctly? Perhaps we should not default FirstConsecutiveStaticIP when VnetSubnetID is set?

I briefly looked at this today, it seems that validation is run before defaulting

validation: [loadApiModel](https://github.com/Azure/acs-engine/blob/bed3f27da4b42f6a446683889f5c54c4c49b87e5/cmd/generate.go#L57) calls [LoadContainerServiceFromFile](https://github.com/Azure/acs-engine/blob/bed3f27da4b42f6a446683889f5c54c4c49b87e5/cmd/generate.go#L135) which calls [DeserializeContainerService](https://github.com/Azure/acs-engine/blob/bed3f27da4b42f6a446683889f5c54c4c49b87e5/pkg/api/apiloader.go#L34) which calls [LoadContainerService](https://github.com/Azure/acs-engine/blob/bed3f27da4b42f6a446683889f5c54c4c49b87e5/pkg/api/apiloader.go#L45) which finally does [validation](https://github.com/Azure/acs-engine/blob/bed3f27da4b42f6a446683889f5c54c4c49b87e5/pkg/api/apiloader.go#L171).
defaulting: [run](https://github.com/Azure/acs-engine/blob/bed3f27da4b42f6a446683889f5c54c4c49b87e5/cmd/generate.go#L61) calls [GenerateTemplate](https://github.com/Azure/acs-engine/blob/bed3f27da4b42f6a446683889f5c54c4c49b87e5/cmd/generate.go#L185) which is setting the defaults [here](https://github.com/Azure/acs-engine/blob/0dececb663cc4e792b786f274690c69ce057f1c0/pkg/acsengine/template_generator.go#L57)

I would be surprised if this worked differently for `acs-engine deploy`.

Reply:
After chatting with @jim-minter, I'll implement the validation unit test and go from there.